astropy/table/info.py

```diff
@@ -0,0 +1,88 @@
+"""
+Table method and functions related to providing information about
+columns and tables.
+"""
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import sys
+import os
+
+import numpy as np
+
+__all__ = ['info']
+
+def info(self, option='attributes', out=''):
+    """
+    Write summary information about column to the ``out`` filehandle.
+    By default this prints to standard output via sys.stdout.
+
+    The ``option` argument specifies what type of information
+    to include.  This can be a string, a function, or a list of
+    strings or functions.  Built-in options are:
+
+    - ``meta``: basic column meta data like ``dtype`` or ``format``
+    - ``stats``: basic statistics: minimum, mean, and maximum
+
+    If a function is specified then that function will be called with the
+    column as its single argument.  The function must return an OrderedDict
+    containing the information attributes.
+
+    If a list is provided then the information attributes will be
+    appended for each of the options, in order.
+
+    Examples
+    --------
+    >>> from astropy.table.table_helpers import simple_table
+    >>> t = simple_table()
+    >>> t['a'].unit = 'm'
+    >>> t.info()
+    <Table length=3>
+    name  dtype  unit
+    ---- ------- ----
+       a   int32    m
+       b float32
+       c string8
+
+    >>> t.info('stats')
+    <Table length=3>
+    name min mean max
+    ---- --- ---- ---
+       a   1  2.0   3
+       b 1.0  2.0 3.0
+       c  --   --  --
+
+    Parameters
+    ----------
+    option: str, function, list of (str or function)
+        Info option (default='attributes')
+    out: file-like object, None
+        Output destination (default=sys.stdout).  If None then the
+        OrderedDict with information attributes is returned
```

Comment:
@taldcroft – I think the `None` option to return the `OrderedDict` isn't implemented yet, right?
```
In [23]: t.info(out=None)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-23-ec9644a5a55b> in <module>()
----> 1 t.info(out=None)

/Users/deil/code/astropy/astropy/table/info.py in info(self, option, out)
     86 
     87     outlines.extend(info.pformat(max_width=-1, max_lines=-1, show_unit=False))
---> 88     out.writelines(outline + os.linesep for outline in outlines)

AttributeError: 'NoneType' object has no attribute 'writelines'
```
I think it would be useful to be able to get a Table or OrderedDict back, so that it's possible to e.g. store the stats.